### DISCUSSION TITLE
[Button] Rendering button fails in test environment

### DISCUSSION THREAD

**Discussion by knoefel** [ariakit/ariakit#1398] [Questions] [answered]
I'm getting the following error when rendering the `<Button />` component inside a `vitest` test environment:

<img width="806" alt="Screenshot 2022-05-23 at 21 11 14" src="https://user-images.githubusercontent.com/2396344/169890035-415de55d-a03a-4105-9fd8-7b7f4623507a.png">

[Referenced Line from Error Message](https://github.com/ariakit/ariakit/blob/da08e9809510b873be5c76122dd240966d9ee504/packages/ariakit-utils/src/events.ts#L150)

I'm a little bit stuck right now - am i missing something obvious? Do i need to mock the `window.frames` property?

Thx in advance for any feedback or tip!!

**Comment by diegohaz** [comment, marked as answer]
I'd say this is probably a bug/lack of support on Vitest. But we can also check window.frames before accessing length here. Would you like to submit a PR?

**Reply by knoefel** [reply]
Sure!
